### DISCUSSION TITLE
[Support] kafka 不能在 onConfigLoad 里修改配置，现在 brokers 列表是从服务发现中心获取的，这个能怎么实现？

### DISCUSSION THREAD

**Discussion by Yuliang-Lee** [midwayjs/midway#3742] [Q&A]
### Describe the problem（描述问题）

brokers 列表是每次服务启动时从服务注册发现中心获取的，数量比较多，不好全部复制出来写死在 config 文件里

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(3.14.7)
✓ @midwayjs/bootstrap(3.14.4)
✓ @midwayjs/bull(3.14.7)
✓ @midwayjs/bull-board(3.14.7)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.14.4)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.14.4)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.14.7)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(not installed)
✖ @midwayjs/kafka(current: 3.15.6, allow: ["3.14.4","3.14.7"])
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.14.7)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.14.7)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.14.7)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(3.14.7)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(not installed)
✓ @midwayjs/ws(not installed)

**Comment by czy88840616** [comment]
你是想启动的时候动态创建 broker 么

**Reply by Yuliang-Lee** [reply]
可以这么理解， 这里 broker 的列表是从远端获取的
